### PR TITLE
feat: UserSkel improvements (`firstname`, `lastname`, `sync`) and Google Auth user information synchronization

### PR DESCRIPTION
**Pull request by phorward**
This should become standard to allow for better pre-defined personalization.

### CODE REVIEW

**Comment by sveneberth** [comment]
Okay for me, but fill a new skel with the information from the google login
https://github.com/viur-framework/viur-core/blob/209396bf72a1fea6fdffac13c825397b50728b71/core/modules/user.py#L448

**Review by sveneberth — CHANGES_REQUESTED** [review]
Please fill them with values, as in https://github.com/viur-framework/viur-core/pull/677

**Comment by phorward** [comment]
> Please fill them with values, as in #677

you mean #2?!

**Comment by phorward** [comment]
@sveneberth I've added a `synced`-Flag to let the user decide if information should be kept synchronous with Google.
I've also changed the way how the `update`-flag is being determined.

File: core/modules/user.py [hunk 1]
```diff
@@ -47,6 +57,16 @@ class UserSkel(skeleton.Skeleton):
         unique=UniqueValue(UniqueLockMethod.SameValue, False, "UID already in use"),
     )
 
+    synced = BooleanBone(
```
Comment (sveneberth) [inline core/modules/user.py:60]:
```suggestion
    sync = BooleanBone(
```
Please rename this bone to `sync`.

File: core/modules/user.py [hunk 2]
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
...
             userSkel["name"] = email
-            isAdd = True
-        else:
-            isAdd = False
-        if isAdd:
+            update = True
+
+        # Take user information from Google, if wanted!
+        if userSkel["synced"]:
+            logging.debug(f"""{userSkel["synced"]=}""")
```
Comment (sveneberth) [inline core/modules/user.py:513]:
```suggestion
        if userSkel["sync"]:
```

File: core/modules/user.py [hunk 3]
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
...
+
+            for target, source in {
+                "name": email,
+                "firstname": userInfo.get("given_name"),
+                "lastname": userInfo.get("family_name"),
+            }.items():
+                logging.debug(f"""{target=} {source=}""")
```
Comment (sveneberth) [inline core/modules/user.py:520]:
```suggestion
```

File: core/modules/user.py [hunk 4]
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
...
+
+                if userSkel[target] != source:
+                    userSkel[target] = source
+                    update = True
+
+        logging.debug(f"{update=}")
```
Comment (sveneberth) [inline core/modules/user.py:526]:
```suggestion
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**